sqlglot/dialects/snowflake.py

```diff
@@ -55,11 +58,11 @@ def _builder(args: t.List) -> exp.Func:
             # Handles `TO_TIMESTAMP(str, fmt)` and `TO_TIMESTAMP(num, scale)` as special
             # cases so we can transpile them, since they're relatively common
             if kind == exp.DataType.Type.TIMESTAMP:
-                if int_value and not safe:
+                if not safe and (int_value or int_scale_or_fmt):
                     # TRY_TO_TIMESTAMP('integer') is not parsed into exp.UnixToTime as
                     # it's not easily transpilable
-                    return exp.UnixToTime(this=value, scale=seq_get(args, 1))
-                if not is_float(value.this):
+                    return exp.UnixToTime(this=value, scale=scale_or_fmt)
```

Comment:
Is `exp.UnixToTime`'s  scale compatible with Snowflake's scale? Should we add any tests for that matter?

Reply:
Yep, the scale units are pretty much the same, not seeing much value in testing this separately for the time being.